### PR TITLE
Added selected connection count

### LINKED ISSUE #592: Show count of selected connections in Push menu
**Is your enhancement related to a problem? Please describe.**

As an offshoot from #589, there's an additional UI improvement we can make to the Push menu in the form of noting how many connections are selected for distribution.

**Describe the solution you'd like**

- [ ] Show the number of selected connections - this needs i18n consideration in whatever string, easier to just put the count in parens rather than trying to pluralize

**Designs**

n/a

**Describe alternatives you've considered**

n/a

**Additional context**

n/a

### PR DESCRIPTION
**Pull request by roshniahuja**
Fixes for issue #592 

### Description of the Change

Issue mentioned URL - https://github.com/10up/distributor/issues/592

Added selected connection count 

Closes

### How to test the Change

To test this feature - Need to select connection from dropdown and it will increase count in Selected connection

### Changelog Entry

> Added - Show count of selected connections in Push menu.

### Credits

Props @roshniahuja

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by roshniahuja** [comment]
@peterwilsoncc,

I have few ESLint errors and I already tried to resolve by installing ESLint and Prettier ESLint extension but still nothing works so If anyone can take a look then it will be appreciated.

**Comment by peterwilsoncc** [comment]
@roshniahuja I've pushed 22a5031fbfb08b1a8f2e91d5cc77ad624926c55f which fixes the linting errors.

Getting the lints set up in your IDE can be luck as much as anything else. If you use VS Code, I can share my extensions with you if you reach out via Slack.

File: assets/js/push.js [hunk 1]
```diff
@@ -661,6 +670,11 @@ jQuery( window ).on( 'load', () => {
 					element.appendChild( removeLink );
 					element.classList = 'added-connection';
 
+					document.querySelector(
+						'.selected-connections-count'
+					).textContent =
+						'( ' + Object.keys( selectedConnections ).length + ' )';
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
						'(' + Object.keys( selectedConnections ).length + ')';
```

Consistent presentation.

File: assets/js/push.js [hunk 2]
```diff
@@ -602,6 +607,10 @@ jQuery( window ).on( 'load', () => {
 				selectedConnections[ type + id ] = dtConnections[ type + id ];
 
 				const element = event.currentTarget.cloneNode( true );
+				document.querySelector(
+					'.selected-connections-count'
+				).textContent =
+					'( ' + Object.keys( selectedConnections ).length + ' )';
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
					'(' + Object.keys( selectedConnections ).length + ')';
```

Consistent presentation

File: assets/js/push.js [hunk 3]
```diff
@@ -581,6 +581,11 @@ jQuery( window ).on( 'load', () => {
 
 				delete selectedConnections[ type + id ];
 
+				document.querySelector(
+					'.selected-connections-count'
+				).textContent =
+					'( ' + Object.keys( selectedConnections ).length + ' )';
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
					'(' + Object.keys( selectedConnections ).length + ')';
```

Consistent presentation

**Review by peterwilsoncc — COMMENTED** [review]
A couple of inline notes about consistency with layout/spacing within the brackets.

I'm wondering if it would be better to use the string `Selected connections (%d)` to ensure the strings are translatable in to all language formats.

The HTML could become:

```html
<header class="with-selected">
	<span class="selected-connections-text"><?php esc_html_e( 'Selected connections', 'distributor' ); ?></span>
	<button class="button button-link selectno-connections unavailable"><?php esc_html_e( 'Clear', 'distributor' ); ?></button>
</header>
```

And the text in `selected-connections-text` replaced in full via the JavaScript. @ravinderk, do you have an opinion on what would be best?

File: assets/js/push.js (hunk 1 above)
Comment (ravinderk) [reply assets/js/push.js]: @peterwilsoncc Yes, I agree with you. We should use `@wordpress/i18n` to make string translatable.

```js
import { sprintf, _n } from '@wordpress/i18n';

sprintf( _n( 'Selected connection (%d)', 'Selected connections (%d)', selectedConnectionsCount, 'text-domain' ), selectedConnectionsCount );
// Selected connection (1)'
// Selected connections (3)'
```

Comment (roshniahuja) [reply assets/js/push.js]: @peterwilsoncc, I have used use @wordpress/i18n to make string translatable. 
CC : @ravinderk

File: assets/js/push.js (hunk 2 above)
Comment (roshniahuja) [reply assets/js/push.js]: Also I have used same as above format but still ESlint error is coming not sure why and also I have used same extension which you have provided for VS Code.

@peterwilsoncc I have maintained consistency for other syntax but still There is one ESlint error coming can you please check.

image

File: assets/js/push.js [hunk 4]
```diff
@@ -580,7 +581,19 @@ jQuery( window ).on( 'load', () => {
 				deleteNode.parentNode.removeChild( deleteNode );
 
 				delete selectedConnections[ type + id ];
-
+				selectedText = sprintf(
+					/* translators: 1) Selected connection content singular name. 2) Selected connection content plural name. */
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
					/* translators: %d: Number of selected connections. */
```

For `_n` translator notes are a little odd, the comment applies to both strings as the placeholders need to be identical.

File: assets/js/push.js [hunk 5]
```diff
@@ -602,6 +615,19 @@ jQuery( window ).on( 'load', () => {
 				selectedConnections[ type + id ] = dtConnections[ type + id ];
 
 				const element = event.currentTarget.cloneNode( true );
+				selectedText = sprintf(
+					/* translators: 1) Selected connection content singular name. 2) Selected connection content plural name. */
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
					/* translators: %d: Number of selected connections. */
```

File: assets/js/push.js [hunk 6]
```diff
@@ -660,6 +686,19 @@ jQuery( window ).on( 'load', () => {
 
 					element.appendChild( removeLink );
 					element.classList = 'added-connection';
+					selectedText = sprintf(
+						/* translators: 1) Selected connection content singular name. 2) Selected connection content plural name. */
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
						/* translators: %d: Number of selected connections. */
```

File: assets/js/push.js [hunk 7]
```diff
@@ -691,6 +730,19 @@ jQuery( window ).on( 'load', () => {
 					);
 
 				delete selectedConnections[ type + id ];
+				selectedText = sprintf(
+					/* translators: 1) Selected connection content singular name. 2) Selected connection content plural name. */
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
					/* translators: %d: Number of selected connections. */
```

File: assets/js/push.js [hunk 8]
```diff
@@ -721,7 +773,18 @@ jQuery( window ).on( 'load', () => {
 			const id = event.currentTarget.getAttribute( 'data-connection-id' );
 
 			delete selectedConnections[ type + id ];
-
+			selectedText = sprintf(
+				/* translators: 1) Selected connection content singular name. 2) Selected connection content plural name. */
```
Comment (peterwilsoncc) [inline assets/js/push.js]:
```suggestion
				/* translators: %d: Number of selected connections. */
```

**Review by peterwilsoncc — COMMENTED** [review]
This looks great, thank you!

One minor note about translator comments for `_n` but otherwise it looks good.

File: assets/js/push.js (hunk 2 above)
Comment (peterwilsoncc) [reply assets/js/push.js]: I see you figured this out. 

Prettier often applies different formatting depending on the length of the line. I've never been able to figure out what the logic is.

File: assets/js/push.js (hunk 5 above)
Comment (roshniahuja) [reply assets/js/push.js]: Comment Updated.

**Review by peterwilsoncc — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**